containertest/docker.go

```diff
@@ -118,54 +103,49 @@ func runContainer(conf *config, pool *dockertest.Pool) (*dockertest.Resource, er
 					1. Run "sudo adduser $USER docker" to add your user to the docker group
 					2. Reboot the machine to make the group membership effective`
 		case strings.Contains(err.Error(), "404"):
-			extraMsg = fmt.Sprintf(". Probably the requested version %q does not exist as a Docker image", conf.mySQLVersion)
+			extraMsg = fmt.Sprintf(". Probably the requested tag %q does not exist as a Docker image", conf.service.ImageTag())
 		}
-		return nil, fmt.Errorf("pool.Run() failed starting mysql container: %w%s", err, extraMsg)
+		return nil, fmt.Errorf("pool.Run() failed starting container: %w%s", err, extraMsg)
 	}
 	return container, nil
 }
 
-// waitUntilUp waits for mysql to be reachable.
-func waitUntilUp(conf *config, pool *dockertest.Pool, container *dockertest.Resource) (string, error) {
-	var outPort string
-	// To get the TCP port number for the mysql server, we have to wait for the docker container to
+// waitUntilUp waits for service to be reachable.
+func waitUntilUp(conf *config, pool *dockertest.Pool, container *dockertest.Resource) (*ConnInfo, error) {
+	var connInfo *ConnInfo
+	// To get the exported TCP port number for the server, we have to wait for the docker container to
 	// actually start, then get the mapped port number.
 	pool.MaxWait = time.Minute
 	if err := pool.Retry(func() error {
-		port := container.GetPort("3306/tcp")
-		if port == "" {
-			return fmt.Errorf("resource.GetPort() returned empty string, container isn't ready yet")
+		notReadyPorts := make([]string, 0, len(conf.service.StartupPorts()))
+		for _, waitPort := range conf.service.StartupPorts() {
+			if waitPort != "" {
+				port := container.GetPort(waitPort)
+				if port == "" {
+					notReadyPorts = append(notReadyPorts, port)
+				}
+			}
+		}
+		if len(notReadyPorts) > 0 {
+			return fmt.Errorf("resource.GetPort() returned empty string for port(s) %+q, container isn't ready yet", notReadyPorts)
```

Comment:
I think we want to wrap this in a retryable error?

Reply:
As in define a new error type called retryable? Is there a point if `dockertest` just swallows it in the retryable case?

Reply:
Sorry - I didn't realize we were using a different retry library. In [go-retry](https://github.com/sethvargo/go-retry), you have to explicitly wrap an error as retryable. I guess `pool.Retry` just retries on all errors?

Reply:
Yeah, it doesn't make any distinction